### ISSUE TITLE
Django admin: make show/hide sections comply to accessibility criteria

### ISSUE THREAD

**Issue by vickyszuchin** [cisagov/manage.get.gov#1950]
### Issue description

Make show/hide sections comply to accessibility criteria. There is an accessibility issue on show/hide sections (h2 should be a button).

See:
-  [slack thread](https://cisa-corp.slack.com/archives/C05BGB4L5NF/p1711644874720319) for analysis and screenshot
- [slack thread](https://cisa-corp.slack.com/archives/C06K96473FA/p1711572361126209) for additional discussion

### Acceptance criteria

- [ ] ensure that show/hide sections confirm to accessibility criteria. It should be a `<button type="button" aria-label="show/hide more information about this section">` , not a `<h2>` with an embedded `<a> `
- [ ] Toggle the aria-label as well as the innerHTML between show/hide as the button is clicked

### Additional context

We should be able to do this sans JS as DJA already toggles the innerHTML for their `<a>`. Just dig through source code.

### Links to other issues

Related to: #1949

**Comment by PaulKuykendall** [comment]
Rachid updated the AC April 4.
